src/browser/ReactDOM.js

```diff
@@ -24,7 +24,7 @@ var mapObject = require('mapObject');
  * @private
  */
 function createDOMFactory(tag) {
-  if (__DEV__) {
+  if (__DEBUG__) {
     return ReactElementValidator.createFactory(tag);
```

Comment:
my recollection is that ReactElementValidator was the thing that does the proptype checking... if so, we dont' want it in debug.